types/puppeteer/index.d.ts

```diff
@@ -1258,7 +1330,7 @@ export interface Page extends EventEmitter, FrameBase {
   deleteCookie(...cookies: DeleteCookie[]): Promise<void>;
 
   /** Emulates given device metrics and user agent. This method is a shortcut for `setUserAgent` and `setViewport`.  */
-  emulate(options: Partial<EmulateOptions>): Promise<void>;
+  emulate(options: EmulateOptions): Promise<void>;
```

Comment:
Can you confirm that the all properties must be present? Last time I checked the api docs where  a bit vague on that front.

Reply:
All properties in `EmulateOptions` are optional already, so the `Partial` monad was redundant.